### DISCUSSION TITLE
Tracking a dependency's branch?

### DISCUSSION THREAD

**Discussion by cpsauer** [renovatebot/renovate#18360] [Request Help] [answered]
Hi wonderful Renovate folks,

Is there a way to autoupdate along a particular branch of a GitHub-sourced dependency? 
(I feel like there has to be, but I'd tried searching around the docs for a good while but couldn't find anything.)

As more background: I'm trying to autoupdate boringssl in a bazel workspace, for which you have to track the master-with-bazel branch. Renovate keeps pulling the dependency back toward new commits on master--even when there are even newer commits on the master-with-bazel branch.

Thanks,
Chris

**Comment by rarkins** [comment, marked as answer]
What is the current `packageFiles with updates` entry for this dependency?

**Reply by cpsauer** [reply]
Thanks for such a quick response, @rarkins.

Here's one where a problematic update is being proposed. The proposed update is to the tip of the master branch from the old tip of the master-with-bazel branch. (And while the new commit was also merged into the master-with-bazel branch, we need to stay on the master-with-bazel branch to get the rest of the changes required for usage from Bazel.)
```json
"datasource": "github-tags",
"depType": "http_archive",
"depName": "boringssl",
"packageName": "google/boringssl",
"currentDigest": "0c34983ad511c6aaa642d994a027d6147fde69e8",
"managerData": {
  "def": "http_archive(\n    name = \"boringssl\",\n    url = \"https://github.com/google/boringssl/archive/0c34983ad511c6aaa642d994a027d6147fde69e8.tar.gz\",\n    sha256 = \"a061b2fedfa9f30bcd5a83ffb67b9fbe4fc30ef5a80482f608aef40f3171f729\",\n    strip_prefix = \"boringssl-0c34983ad511c6aaa642d994a027d6147fde69e8\",\n)"
},
"depIndex": 9,
"updates": [
  {
    "updateType": "digest",
    "newDigest": "14aa0de18f638a92be13597bc1b8a95ca8fcf8a4",
    "branchName": "renovate/boringssl-digest"
  }
],
"warnings": [],
"versioning": "semver"
```

**Reply by rarkins** [reply]
How is Renovate meant to know that it should track the master-with-bazel branch? I don't see it referenced anywhere, so do you expect it will hunt to find a branch where the current SHA lives?

**Reply by cpsauer** [reply]
Hence the question!

Ideally, it'd be great if renovate updated to the (newest) tip of the branch(es) the current commit is on. But I'd also be happy to specify the branch manually.

I take it that there's no current way to configure this?

**Reply by rarkins** [reply]
I don't think there's any way out of the box to do this today. It _might_ be nearly possible by using the `git-refs` datasource with `currentValue` set to the branch name, but for that you need to use the regex manager instead of bazel, which would in turn introduce a problem that the regex manager may not be able to update all the references you need (the SHA is in more than one location, for example).

Once @zharinov has finished some current refactoring of bazel parsing, we might be able to introduce a convention which lets you define the branch name within the file itself, which would be more ideal than needing to configure it manually in Renovate config.

**Reply by cpsauer** [reply]
Got it. Bummer, but thanks for letting me know--and for all the great stuff you do!

What would be the best way to track? Would you like me to file a feature request? Or is there an easy way for you to migrate this discussion or add it to an existing issue?

The best workaround I can think of in the meantime is to configure a pull bot to create a mirror of the original repo's branch, and then have renovate track that.

**Reply by rarkins** [reply]
Yes, go ahead with a feature request. We'd want something similar to how we allow pinning to a tag for the `github-actions` manager, although sounds like you need the ability to pin to a ref (branch), not tag.
